src/Views/AppInfoView.vala

```diff
@@ -252,30 +252,45 @@ namespace AppCenter.Views {
 
             var project_license = package.component.project_license;
             if (project_license != null) {
-                string license_url = "https://choosealicense.com/licenses/";
-                switch (project_license) {
-                    case "Apache-2.0":
-                        license_url = license_url + "apache-2.0";
-                        break;
-                    case "GPL-2":
-                    case "GPL-2.0":
-                    case "GPL-2.0+":
-                        license_url = license_url + "gpl-2.0";
-                        break;
-                    case "GPL-3":
-                    case "GPL-3.0":
-                    case "GPL-3.0+":
-                        license_url = license_url + "gpl-3.0";
-                        break;
-                    case "LGPL-2.1":
-                    case "LGPL-2.1+":
-                        license_url = license_url + "lgpl-2.1";
-                        break;
-                    case "MIT":
-                        license_url = license_url + "mit";
-                        break;
+                string? license_copy;
+                string? license_url;
+                if (project_license.has_prefix ("LicenseRef")) {
+                    // i.e. `LicenseRef-proprietary=https://example.com`
+                    var split_license = project_license.split_set ("=", 2);
+
+                    var license_type = split_license[0].split_set ("-", 2)[1];
+                    var pretty_license_type = license_type.substring (0, 1).up () + license_type.substring (1);
```

Comment:
This feels the most dirty and language-specific; it uppercases the `proprietary` string, or whatever comes after the `LicenseRef-` and before the `=`. In practice, this works fine in Pop!_Shop in English, but it's not translatable and might be trying to be too clever.